tools/ontology/codegen/src/main/java/org/visallo/tools/ontology/ingest/codegen/CodegenCLI.java

```
@@ -0,0 +1,300 @@
+                    }
+                    return property;
+                })
+                .collect(Collectors.toList());
+    }
+
+    private String constantNameFromClassName(String className) {
+        String[] classNameParts = StringUtils.splitByCharacterTypeCamelCase(className);
+        return Arrays.stream(classNameParts)
+                .map(String::toUpperCase)
+                .collect(Collectors.joining("_"));
+    }
+
+
+    private String classNameFromIri(String iri) {
+        Matcher matcher = IRI_FORMAT.matcher(iri);
+        if (matcher.matches()) {
+            String[] classNameParts = StringUtils.splitByCharacterTypeCamelCase(matcher.group(2));
+            return Arrays.stream(classNameParts)
+                    .filter(classNamePart -> CLASSNAME_PART_MATCHER.matcher(classNamePart).matches())
+                    .map(StringUtils::capitalize)
+                    .collect(Collectors.joining(""));
+        } else {
+            LOGGER.error("Unsupported iri pattern %s", iri);
+        }
+        return null;
+    }
+
+    private String packageNameFromIri(String iri) {
+        Matcher matcher = IRI_FORMAT.matcher(iri);
+        if (matcher.matches()) {
+            String[] baseIriParts = matcher.group(1).split("/", -1);
+
+            String[] packageParts = baseIriParts[0].split("\\.", -1);
+            ArrayUtils.reverse(packageParts);
+            if (baseIriParts.length > 1) {
+                packageParts = (String[]) ArrayUtils.addAll(packageParts, ArrayUtils.subarray(baseIriParts, 1, baseIriParts.length));
+            }
+
+            String packageName = String.join(".", packageParts);
+            if (packageName.toLowerCase().equals("org.w3.www.2002.07.owl")) {
+                packageName = "org.w3.www.owl";
+            }
+            return packageName;
+        } else {
+            LOGGER.error("Unsupported iri pattern %s", iri);
```

Do we want to throw an exception here?